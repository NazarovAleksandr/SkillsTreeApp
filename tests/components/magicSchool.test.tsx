import * as React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import MagicSchool from '../../src/components/magicSchools/magicSchool';

describe('MagicSchool', () => {
    test('Renders passive display', () => {
        let school = {name: 'testSchool', id: '1'};
        const component = shallow(<MagicSchool school={school}></MagicSchool>);

        expect(component).toMatchSnapshot('passiveDisplay');
    });
    test('Renders active school', () => {
        let school = {name: 'testSchool', id: '1'};
        const component = shallow(<MagicSchool school={school} isActive={true}></MagicSchool>);

        expect(component).toMatchSnapshot('activeDisplay');
    });
    test('Renders is edition state', () => {
        let school = {name: 'testSchool', id: '1'};
        const component = shallow(<MagicSchool school={school} inEdition={true}></MagicSchool>);

        expect(component).toMatchSnapshot('inEditionState');
    });
    test('Calls select callback on click', () => {
        let school = {name: 'testSchool', id: '1'};
        const fn = jest.fn()

        const component = shallow(<MagicSchool school={school} inEdition={true} onSelect={fn}></MagicSchool>);
        component.find('.school').simulate('click');

        expect(fn).toBeCalled();
    });
    test('Calls callback on input bluring', () => {
        let school = {name: 'testSchool', id: '1'};
        const fn = jest.fn()

        const component = shallow(<MagicSchool school={school} inEdition={true} onEditionEnd={fn}></MagicSchool>);
        component.find('.input').simulate('blur');

        expect(fn).toBeCalled();
    });
});