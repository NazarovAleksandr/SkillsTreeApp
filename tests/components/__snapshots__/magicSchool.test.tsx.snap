// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicSchool Renders active school: activeDisplay 1`] = `
<div
  className="school active"
>
  <span
    className="title"
    title="testSchool"
  >
    testSchool
  </span>
</div>
`;

exports[`MagicSchool Renders is edition state: inEditionState 1`] = `
<div
  className="school"
>
  <div
    className="active-container"
  >
    <input
      autoFocus={true}
      className="input"
      defaultValue="testSchool"
    />
  </div>
</div>
`;

exports[`MagicSchool Renders passive display: passiveDisplay 1`] = `
<div
  className="school"
>
  <span
    className="title"
    title="testSchool"
  >
    testSchool
  </span>
</div>
`;
